system/ee/legacy/libraries/Template.php

```diff
@@ -403,6 +403,8 @@ public function parse(&$str, $is_embed = false, $site_id = '', $is_layout = fals
                             'Snippet "' . $variable . '"'
                         );
 
+                        $replace = $this->addSourceMarker($replace, $variable, $added_globals['template_type']);
```

Comment:
Because this has potential to break templates (e.g. comment inside a comment, or when partial is used as a parameter for HTML tag), we need more precise control here.

At the minimum, we need to add config override and only add markers when it's set. Ideally, we might want to be able to control this on template level, like we do with front-end editing tags.

Or make it better code overall, aware of context and not break things